### DISCUSSION TITLE
PreventGestureRecognition

### DISCUSSION THREAD

**Discussion by workgroupengineering** [AvaloniaUI/Avalonia#13177] [Q&A] [answered]
It's a curiosity. Why when I need to implement a custom `GestureRecognizer` is it not enough to set Handle to true to prevent recognition from other `GestureRecognizer`, but I need to call the `PreventGestureRecognition` method?

**Comment by maxkatz6** [comment]
cc @emmauss @kekekeks

**Reply by kekekeks** [reply]
Looks like an oversight.

**Reply by emmauss** [reply]
`PreventGestureRecognition` prevents other gestures from receiving that event, but not the source element, or any elements in the route. Setting `Handled` to `true` would prevent any control from handling it including the parent controls. Gestures are always run, even if the event is handled.
Take the Button as an example. It handled `PointerPressed` and sets `Handled` to true. This would normally prevent `ScrollViewer` from handling the ScrollGesture, but as Gestures are always run on handled events, `ScrollGestureRecognizer` will run after Button handles the event. You should still be able to scroll even if you are touching a Button. To prevent another gesture from handling pointer events after the `ScrollGestureRecognizer`, e.g. a parent control hosting a `PullGestureRecognizer`, `PreventGestureRecognition` is called by `ScrollGestureRecognizer`.

**Reply by kekekeks** [reply]
I think we should auto-call PreventGestureRecognition from GestureRecognizer.Capture

**Reply by emmauss** [reply]
Capture doesn't have access to the event args

**Reply by emmauss** [reply, marked as answer]
https://github.com/AvaloniaUI/Avalonia/pull/13323 @kekekeks
